src/Tree/Tree.js

```diff
@@ -1,238 +1,330 @@
-import { Dropdown } from '../Dropdown/Dropdown';
+import classnames from 'classnames';
 import PropTypes from 'prop-types';
+import shortid from '../utils/shortId';
 import React, { Component } from 'react';
 
-export class Tree extends Component {
+export class TreeCell extends Component {
+    render() {
+        const {
+            children,
+            className,
+            ...rest
+        } = this.props;
+
+        const cellClassName = classnames(
+            'fd-tree__col',
+            className
+        );
+
+        return (
+            <div
+                {...rest}
+                className={cellClassName}>
+                {children}
+            </div>
+        );
+    }
+}
+
+TreeCell.propTypes = {
+    children: PropTypes.node,
+    className: PropTypes.string
+};
+
+export class TreeHeader extends Component {
+    render() {
+        const {
+            buttonProps,
+            children,
+            isExpanded,
+            onExpandAll,
+            ...rest
+        } = this.props;
+
+        return (
+            <div {...rest} className='fd-tree fd-tree--header'>
+                <div className='fd-tree__row fd-tree__row--header'>
+                    {
+                        React.Children.map(children, (child, index) => {
+                            const isFirstTreeCell = index === 0 && child.type && child.type.name === 'TreeCell';
+
+                            // Add control class to first TreeCell element
+                            const className = classnames({
+                                'fd-tree__col--control': isFirstTreeCell
+                            });
+
+                            // Add expand button to first TreeCell element
+                            const newChildren = isFirstTreeCell ? (
+                                <div>
+                                    <button
+                                        {...buttonProps}
+                                        aria-label='expand'
+                                        aria-pressed={isExpanded}
+                                        className='fd-tree__control'
+                                        onClick={onExpandAll} />
+                                    {child.props && child.props.children}
+                                </div>
+                            ) : child.props && child.props.children;
+
+                            return React.cloneElement(child, {
+                                children: newChildren,
```

Comment:
This seems dangerous.  If we're providing semantic components for them to use, we probably shouldn't be messing around with `children`.

Reply:
I don't like it either. But with the first cell in each row, there needs to be the expand/collapse button within the cell div, and I'm not sure how to achieve that without setting children. I tried pulling the button outside of the cell div as a sibling element and the styles didn't look right.